core/src/value/deserialize.rs

```diff
@@ -540,4 +687,143 @@ mod tests {
         assert!(parse_datetime(b"20171130101010.204+01", default_offset).is_err());
         assert!(parse_datetime(b"20171130101010.204+011", default_offset).is_err());
     }
+    #[test]
+    fn test_parse_date_range() {
```

Comment:
Only checking for whether it parses OK gives very little confidence. Please make more specific assertions over the outcome of the operation. Where relevant, this can also be done for errors using `assert!` and `matches!` (as they are also part of the API).

Example:

```rust
assert_eq!(
    parse_date_range("1914-".as_bytes()).unwrap(),
    (NaiveDate::ymd(1914, 1, 1), None),
);
```